locator/topology.hh

```diff
@@ -114,15 +217,30 @@ public:
     void sort_by_proximity(inet_address address, inet_address_vector_replica_set& addresses) const;
 
 private:
+    using mutable_node_ptr = lw_shared_ptr<node>;
+
     // default constructor for cloning purposes
-    topology() = default;
+    topology() noexcept;
+
+    node_ptr add_node(mutable_node_ptr node);
+    void remove_node(node_ptr node);
+    void do_remove_node(mutable_node_ptr node);
 
     /**
      * compares two endpoints in relation to the target endpoint, returning as
      * Comparator.compare would
      */
     int compare_endpoints(const inet_address& address, const inet_address& a1, const inet_address& a2) const;
 
+    unsigned _shard;
+    std::unordered_set<mutable_node_ptr> _all_nodes;
+    node_ptr _local_node;
```

Comment:
Since local means "local to the datacenter", let's call it _this_node.

Reply:
sounds good

Reply:
renamed in v8